apache/superset
# feat(explore): Implement data panel redesign

### SUMMARY
This PR implements the redesign of data panel in Explore (below the chart container).

1. Removed the "Data" header that expands/collapses on click
2. Moved expand/collapse functionality to tabs header. When collapsed, clicking on expand button opens tabs showing "Results" by default OR the last active tab before collapsing
3. When collapsed, user can also click on tab name to expand with that tab active
4. When expanded, user can hide the tables by clicking on collapse button OR by clicking on active tab name
5. Moved row count, copy button and search from tabs header to tab content - now they look more in context of currently displayed table
6. Now Results and Samples have separate searches - the search text is no longer shared between those 2 tables
7. Refactored ChartPanel - now we no longer manually calculate widths and heights of chart panel and tables panel. It's all based on css flex.
8. Tables functionality remains unchanged - the request is sent only when the table is activated the first time or there were changes to query

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
<img width="1786" alt="image" src="https://user-images.githubusercontent.com/15073128/163774386-0af6c816-48bf-4362-be70-0cf36fb19054.png">
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/15073128/163774419-642c9a8c-02e2-483f-b433-a6967f0b8c6f.png">

After:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/15073128/163785258-aa293ede-90e4-4d63-950b-5309f41cc3ff.png">
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/15073128/163774710-9967adf7-6a0f-47a9-9d74-a24554f255f2.png">


### TESTING INSTRUCTIONS
1. Create a chart
2. Verify that collapsing/expanding works as described in PR's summary
3. Verify that copy button and row counter work as before for both results and samples
4. Verify that typing search text in 1 table doesn't affect search filter in second table
5. Verify that chart panel and data table panel don't overflow on each other and scale well when window width changes

### ADDITIONAL INFORMATION



CC @kasiazjc
